src-ui/lib/components/BooksTable.svelte

```diff
@@ -56,7 +58,18 @@
   }
 </script>
 
-<svelte:document on:scan={handleScan} use:listenForBarcodes />
+
+<button on:click={() => onScan.simulate(document, '1234567890123')}>
+	Simulate ISBN
+</button>
+
+{#await promise}
```

Comment:
I might move the loading state out into it's own variable (isLoading) with a default value of `false`, and then update that value based on where in the process the addBook promise is. Having an explicit `isLoading` variable is easier to manage/reason about than relying on the promise here, imo.

Reply:
The `#await` block is a built-in Svelte pattern. It's not worse than using `isLoading`. The only question is whether there are other things that need to rely on the state; in which case we can expose that state. I would explore existing open source svelte code on Github to see if there are patterns to learn from.